### ISSUE TITLE
Suggestions for static code analysis  for ECS

### ISSUE THREAD

**Issue by micz84** [JetBrains/resharper-unity#1321]
I have some ideas for code ECS analysis that ReSharper may support. 
-Native Containers disposing
-Checks if JobHandle is passed to CommandBufferSystem using AddJobHandleForProducer(handle)

**Comment by Lattyware** [comment]
Agreed, ECS has a number of easy pitfalls which some static analysis would really help on. Alongside what you have suggested, I would also suggest:

    - In IJobForEach based jobs, suggest ReadOnly and WriteOnly attributes as appropriate.
    - In IJobForEach based jobs, suggest moving components not read from or written to a RequireComponentTag attribute (not using it or ReadOnly is an error for completely empty components).
